Tools/autotest/arduplane.py

```diff
@@ -3958,6 +3958,96 @@ def MidAirDisarmDisallowed(self):
         self.disarm_vehicle(force=True)
         self.reboot_sitl()
 
+    def AerobaticsScripting(self):
+        '''Fixed Wing Aerobatics'''
+        self.start_subtest("Aerobatics Scripting")
+        ex = None
+        applet_script = "Aerobatics/FixedWing/plane_aerobatics.lua"
+        trick72 = "Aerobatics/FixedWing/trick72.txt"
+        messages = []
+
+        def my_message_hook(mav, message):
+            if message.get_type() != 'STATUSTEXT':
+                return
+            messages.append(message)
+        self.install_message_hook(my_message_hook)
+
+        model = "plane-3d"
+
+        self.customise_SITL_commandline(
+            [],
+            model=model,
+            defaults_filepath="",
+            wipe=True)
+
+        try:
+            self.install_applet_script(applet_script)
+            self.install_applet_script(trick72)
+            self.reboot_sitl()
+            self.delay_sim_time(10)
+        except Exception as e:
+            self.print_exception_caught(e)
+            ex = e
+
+        self.context_push()
+        self.set_parameter("TRIK_ENABLE", 1)
+        self.set_rc(7, 1000)
+
+        self.scripting_restart()
+        self.set_parameters({
+            "TRIK1_ID": 72,
+            "RC7_OPTION" : 300,
+            "RC9_OPTION" : 301,
+        })
+
+        self.wait_ready_to_arm()
+        self.change_mode("TAKEOFF")
+        self.arm_vehicle()
+        self.wait_altitude(30, 40, timeout=30, relative=True)
+        self.change_mode("CRUISE")
+
+        self.context_collect('STATUSTEXT')
+        self.set_rc(9, 1000)
+        self.delay_sim_time(1)
+        self.set_rc(7, 1500)
+
+        self.wait_text("Trick 1 selected (SuperAirShow)", check_context=True)
+        self.set_rc(7, 2000)
+        self.wait_text("Trick 1 started (SuperAirShow)", check_context=True)
+
+        highest_error = 0
+        while True:
+            m = self.mav.recv_match(type='NAMED_VALUE_FLOAT', blocking=True, timeout=2)
+            if not m:
+                break
```

Comment:
```suggestion
            m = self.assert_receive_named_value_float("YOUR_NAMED_VALUE_FLOAT_NAME_HERE", timeout=2)
```

You're not filtering to a single named value float here, so if we start sending more random instances (like `AS2`) then we might break this test.

Reply:
fixed